src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -476,69 +501,52 @@ protected override int ExecuteTool(string pathToTool, string responseFileCommand
+                logger.Log($"CommandLine = '{commandLineCommands}'");
+                logger.Log($"BuildResponseFile = '{responseFileCommands}'");
 
-                    var clientDir = Path.GetDirectoryName(PathToManagedTool);
-                    if (clientDir is null || tempDir is null)
-                    {
-                        return base.ExecuteTool(pathToTool, responseFileCommands, commandLineCommands);
-                    }
-
-                    // Note: we can't change the "tool path" printed to the console when we run
-                    // the Csc/Vbc task since MSBuild logs it for us before we get here. Instead,
-                    // we'll just print our own message that contains the real client location
-                    Log.LogMessage(ErrorString.UsingSharedCompilation, clientDir);
-
-                    var buildPaths = new BuildPathsAlt(
-                        clientDir: clientDir,
-                        workingDir: workingDir,
-                        // MSBuild doesn't need the .NET SDK directory
-                        sdkDir: null,
-                        tempDir: tempDir);
-
-                    // Note: using ToolArguments here (the property) since
-                    // commandLineCommands (the parameter) may have been mucked with
-                    // (to support using the dotnet cli)
-                    var responseTask = BuildServerConnection.RunServerCompilationAsync(
-                        Language,
-                        RoslynString.IsNullOrEmpty(SharedCompilationId) ? null : SharedCompilationId,
-                        GetArguments(ToolArguments, responseFileCommands).ToList(),
-                        buildPaths,
-                        keepAlive: null,
-                        libEnvVariable: LibDirectoryToUse(),
-                        logger: logger,
-                        cancellationToken: _sharedCompileCts.Token);
-
-                    responseTask.Wait(_sharedCompileCts.Token);
-
-                    var response = responseTask.Result;
-                    if (response != null)
-                    {
-                        ExitCode = HandleResponse(response, pathToTool, responseFileCommands, commandLineCommands, logger);
-                    }
-                    else
-                    {
-                        logger.LogError($"Server compilation failed, falling back to {pathToTool}");
-                        Log.LogMessage(ErrorString.SharedCompilationFallback, pathToTool);
-
-                        ExitCode = base.ExecuteTool(pathToTool, responseFileCommands, commandLineCommands);
-                    }
+                var clientDir = Path.GetDirectoryName(PathToManagedTool);
+                if (clientDir is null || tempDir is null)
+                {
+                    LogCompilationMessage(logger, CompilationKind.Tool, $"using command line tool because we can't find client directory {PathToManagedTool}");
```

Comment:
```suggestion
                    LogCompilationMessage(logger, CompilationKind.Tool, $"using command line tool because we could not find client directory '{PathToManagedTool}'");
```